packages/react/src/menu/root/useMenuRoot.ts

```diff
@@ -22,11 +22,12 @@ import { useOpenChangeComplete } from '../../utils/useOpenChangeComplete';
 import type { TextDirection } from '../../direction-provider/DirectionContext';
 import { useScrollLock } from '../../utils/useScrollLock';
 import {
-  type OpenChangeReason,
+  type BaseOpenChangeReason,
   translateOpenChangeReason,
 } from '../../utils/translateOpenChangeReason';
 
 const EMPTY_ARRAY: never[] = [];
+export type MenuOpenChangeReason = BaseOpenChangeReason | 'item-select' | 'cancel-open';
```

Comment:
These types should be exported from the package so they can be used in userland functions